### PR TITLE
Updates accredited provider data

### PR DESCRIPTION
**Pull request by dcyoung-dev**
## Context

We have identified Providers which have incorrect Accredited Provider data. 

## Changes proposed in this pull request

We will update the Accredited Provider for given Providers and their Courses for the 2024 Recruitment Cycle
Requires #4589 

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated added to the Azure KeyVault
- [ ] Inform data insights team due to database changes
- [ ] Make sure all information from the Trello card is in here
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Attach PR to Trello card

### CODE REVIEW

**Comment by dcyoung-dev** [comment]
https://trello.com/c/Oi7GdhWA

File: db/data/20241011081455_update_accredited_providers.rb
Comment (defong) [inline db/data/20241011081455_update_accredited_providers.rb:6]: Pointless comments, since it will, perhaps make `AccreditedProviderUpdater` reflect it instead.

File: app/services/publish/accredited_provider_updater.rb
Comment (defong) [inline app/services/publish/accredited_provider_updater.rb:32]:
```suggestion
      courses = provider.courses
```

File: app/services/publish/accredited_provider_updater.rb
Comment (defong) [inline app/services/publish/accredited_provider_updater.rb:33]:
```suggestion
      courses.update_all!(accredited_provider_code: new_accredited_provider.provider_code)
```

https://apidock.com/rails/ActiveRecord/Relation/update_all

File: app/services/publish/accredited_provider_updater.rb
Comment (defong) [inline app/services/publish/accredited_provider_updater.rb:25]: May need to guard against have 2 lots of 
```ruby
[
  {
    UcasProviderCode: 'CKY',
    Description: 'New'
  },
  {
    UcasProviderCode: 'CKY',
    Description: 'Existing'
  }
]

```

File: db/data/20241011081455_update_accredited_providers.rb
Comment (defong) [inline db/data/20241011081455_update_accredited_providers.rb:14]: Looks right to me

File: db/data/20241011081455_update_accredited_providers.rb
Comment (defong) [inline db/data/20241011081455_update_accredited_providers.rb:35]: Can it be deleted?

File: spec/services/publish/accredited_provider_updater_spec.rb
Comment (defong) [inline spec/services/publish/accredited_provider_updater_spec.rb:16]: Prehaps find_or_create

File: db/data/20241011081455_update_accredited_providers.rb
Comment (defong) [inline db/data/20241011081455_update_accredited_providers.rb:63]: missing 2025?

See 
https://gist.github.com/avinhurry/ab64d4dc83dbc018b1dbda1eb2195624#file-2as-rb-L40-L71

File: spec/services/publish/accredited_provider_updater_spec.rb
Comment (defong) [inline spec/services/publish/accredited_provider_updater_spec.rb:12]: provider code is 3 chars

File: spec/services/publish/accredited_provider_updater_spec.rb
Comment (defong) [inline spec/services/publish/accredited_provider_updater_spec.rb:14]: provider code is 3 chars

**Review by defong — CHANGES_REQUESTED** [review]
Seems to of missed out `2AS` for `2025` can you confirm that it is correct?

File: db/data/20241011081455_update_accredited_providers.rb
Comment (dcyoung-dev) [reply db/data/20241011081455_update_accredited_providers.rb:6]: No these are only here to match up and explain what we're trying to do. 

These comments are extracted from the Trello card to avoid having to jump between 2 sources. 

The comments will be removed before merging

**Comment by inulty-dfe** [comment]
This PR moves **all** the providers' courses to be accredited by the new accredited provider.
This PR does not update the user permissions of the providers. If this needs to be done, the providers should review their users permissions.

**Pull request merged**
